src/com.unity.git.api/Api/Git/Tasks/GitResetTask.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Threading;
+
+namespace Unity.VersionControl.Git.Tasks
+{
+    public class GitResetTask : ProcessTask<string>
+    {
+        private const string TaskName = "git reset";
+        private readonly string arguments;
+
+        public GitResetTask(string changeset, GitResetMode resetMode,
+            CancellationToken token, IOutputProcessor<string> processor = null)
+            : base(token, processor ?? new SimpleOutputProcessor())
+        {
+            var mode = "";
+            mode +=
+                resetMode == GitResetMode.NonSpecified ? "" :
```

Comment:
probably more readable to put this logic in a separate private method

Reply:
Fixed.